sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -38,30 +39,41 @@ public static Hub create(final @NotNull SentryOptions options) {
       // creates Hub working in server-side session mode
       final SessionAggregates sessionAggregates =
           new SessionAggregates(options.getRelease(), options.getEnvironment());
-      sessionAggregates.start();
       final ServerSessionManager sessionTracker = new ServerSessionManager(sessionAggregates);
       final ISentryClient client = new SentryClient(options, sessionTracker);
       final Stack stack = new Stack(options.getLogger(), new StackItem(options, client, scope));
-      return new Hub(options, stack, sessionTracker, sessionAggregates);
+      return new Hub(
+          options,
+          stack,
+          sessionTracker,
+          sessionAggregates,
+          new SessionsFlusher(sessionAggregates, client));
     } else {
       // creates Hub working in client-side session mode
       final ISentryClient client = new SentryClient(options, new ClientSessionUpdater(options));
       final Stack stack = new Stack(options.getLogger(), new StackItem(options, client, scope));
-      return new Hub(options, stack, new ClientSessionTracker(options, stack), null);
+      return new Hub(options, stack, new ClientSessionTracker(options, stack), null, null);
     }
   }
 
   private Hub(
       final @NotNull SentryOptions options,
       final @NotNull Stack stack,
       final @NotNull SessionTracker sessionTracker,
-      final @Nullable SessionAggregates sessionAggregates) {
+      final @Nullable SessionAggregates sessionAggregates,
+      final @Nullable SessionsFlusher sessionFlusher) {
     this.options = options;
     this.stack = stack;
     this.tracesSampler = new TracesSampler(options);
     this.sessionTracker = sessionTracker;
     this.lastEventId = SentryId.EMPTY_ID;
     this.sessionAggregates = sessionAggregates;
+    if (sessionFlusher != null) {
+      this.sessionFlusher = sessionFlusher;
+      this.sessionFlusher.start();
+    } else {
+      this.sessionFlusher = null;
```

Comment:
Do we need `else` here at all?

Reply:
`sessionFlusher` has to be initialized, so yes we do.